src/nvim/window.c

```diff
@@ -1939,6 +1941,16 @@ int win_close(win_T *win, bool free_buf)
       return FAIL;
   }
 
+  // fire WinClosed event just before freeing memory associated with window
+  if (has_event(EVENT_WINCLOSED)) {
```

Comment:
should also check `!recursive` (see other comments).

should be skipped if `recursive` (see other comments).

```c
if (!recursive && has_event(EVENT_WINCLOSED)) {
  ...
```

Reply:
I said before, I do not see where/how is `recursive` defined.